books/projects/curve25519/ecp.lisp

```diff
@@ -106,7 +106,7 @@
   :hints (("Goal" :use ((:instance mod*rewrite-2 (a (* a b)) (b c))))))
 
 (defthm mod*rewrite-4
-  (implies (and (integerp a) (integerp b) (integerp c))
+  (implies (and (integerp a) (integerp b) (integerp c) (integerp d))
```

Comment:
How come you added the extra hypothesis?  Typically extra hypotheses make a rule harder to use?  Or maybe we actually need it to be harder to use in this case, so that one branch of rewriting gets cut short?